actionkit/api/mailing.py

```diff
@@ -0,0 +1,154 @@
+import json
+import re
+import requests
+import sys
+
+from actionkit.api.base import ActionKitAPI
+
+try:
+    import urllib.parse
+    encode_url = urllib.parse.urlencode
+except ImportError:
+    # python2
+    import urllib
+    encode_url = urllib.urlencode
+
+class AKMailingAPI(ActionKitAPI):
+
+    def update_mailing(self, mailing_id, update_dict):
+        res = self.client.patch(
+            #the '/' at the end is IMPORTANT!
+            '%s/rest/v1/mailing/%s/' % (self.base_url, mailing_id),
+            data=json.dumps(update_dict))
+        return self._http_return(res)
+
+    def get_mailings(self, mailing_id=False, query_params={}):
+        if getattr(self.settings, 'AK_TEST', False):
+                return TEST_DATA.get(mailing_id)
+        if mailing_id:
+            res = self.client.get(
+                #the '/' at the end is IMPORTANT!
+                '%s/rest/v1/mailing/%s/' % (self.base_url, mailing_id))
+            return {'res': res,
+                    'mailing': res.json() if res.status_code == 200 else None}
+        elif query_params:
+            res = self.client.get(
+                #the '/' at the end is IMPORTANT!
+                '%s/rest/v1/mailing/?%s' % (self.base_url, encode_url(query_params)))
```

Comment:
oh, also you don't need to do this -- `requests.get` has a `params` argument:
https://2.python-requests.org/en/master/api/#requests.get

Reply:
Ah, makes sense. Other api interfaces in this library are using url_encode - they should probably be updated.